src/rust/plugin-work-queue/src/client.rs

```diff
@@ -40,62 +53,87 @@ where
     }
 
     /// Adds a new execution job for a generator
+    #[tracing::instrument(skip(self, request), err)]
```

Comment:
The `tracing::instrument` macro serves two (main) purposes:

1. To add arguments and return values to the trace
2. To instrument asynchronous functions so that traces within them are correct - otherwise you end up with a single trace that has events from multiple executing futures

In general I would say 'skip anything you do not want to add as context', and generally I would say skip everything and then selectively add things back.

Not sure what you mean re: macro.